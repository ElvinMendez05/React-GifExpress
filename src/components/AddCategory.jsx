import { useState } from "react";
import PropTypes from "prop-types";

// eslint-disable-next-line react/prop-types
export const AddCategory = ({onNewCategory}) => {
  
    const [inputValue, setInputValue] = useState('Dragon ball');
    
    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;

        setInputValue('');
        onNewCategory(inputValue.trim());
    }
    
    return (
       <div className="contentForm">
           <form onSubmit={onSubmit} aria-label="form"> 
                <input 
                    type="text" 
                    placeholder="Buscar gifs"
                    onChange={onInputChange}   
                />
                <button type="submit" value={inputValue}> Buscar </button>
            </form>   
        </div>
   );
};

AddCategory.PropTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
