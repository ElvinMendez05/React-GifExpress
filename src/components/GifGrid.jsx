import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

// eslint-disable-next-line react/prop-types
export const GifGrid = ({category}) => {
   
   const {images, isLoading} = useFetchGifs(category) 

    return (
        <>
          <h1>{category}</h1>
         
         { isLoading && (<h2>Loading...</h2>) }
         
          <div className="card-grid">
              {
                images.map( (image)  => (
                    <GifItem 
                      key={image.id}
                      {...image} //captura todos los items 
                    />  
                ))  
              }
          </div>
        </>
    )
}

GifGrid.prototype = {
  category: PropTypes.string.isRequired
}