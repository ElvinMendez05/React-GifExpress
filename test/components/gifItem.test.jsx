import { GifItem} from "../../src/components/GifItem";
import { render, screen} from "@testing-library/react";

/* eslint-disable no-undef */
describe('Test in the component', () => {
   
    const title = 'Goku';
    const url = 'http://api//dragon';

    test('should match the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test("Debe mostrar la imagen con el url y el alt", ()=> {
        render(<GifItem title={title} url={url}/>);
        
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test("Test in this compoent", ()=> {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
   
});

