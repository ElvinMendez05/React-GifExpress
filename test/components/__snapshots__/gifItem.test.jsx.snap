// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://api//dragon"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;

exports[`Test  should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://api//dragon"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;

exports[`Test in the component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://api//dragon"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
